File: ios/VllowDugout/Models/UserProfile.swift
import Foundation
import SwiftData

@Model
final class UserProfile {
    var id: UUID
    var displayName: String
    var avatarURL: String
    var vCoins: Int
    var winStreak: Int
    var globalRank: Int
    var drillsCompleted: Int
    var totalTrainingMinutes: Int
    var joinDate: Date

    init(
        displayName: String = "Champion",
        avatarURL: String = "",
        vCoins: Int = 1250,
        winStreak: Int = 12,
        globalRank: Int = 422,
        drillsCompleted: Int = 87,
        totalTrainingMinutes: Int = 2340
    ) {
        self.id = UUID()
        self.displayName = displayName
        self.avatarURL = avatarURL
        self.vCoins = vCoins
        self.winStreak = winStreak
        self.globalRank = globalRank
        self.drillsCompleted = drillsCompleted
        self.totalTrainingMinutes = totalTrainingMinutes
        self.joinDate = Date()
    }
}

File: flutter_app/lib/models/user_profile.dart
import 'package:uuid/uuid.dart';

class UserProfile {
  final String id;
  String displayName;
  String avatarURL;
  int vCoins;
  int winStreak;
  int globalRank;
  int drillsCompleted;
  int totalTrainingMinutes;
  final DateTime joinDate;

  UserProfile({
    String? id,
    this.displayName = 'Champion',
    this.avatarURL = '',
    this.vCoins = 1250,
    this.winStreak = 12,
    this.globalRank = 422,
    this.drillsCompleted = 87,
    this.totalTrainingMinutes = 2340,
    DateTime? joinDate,
  })  : id = id ?? const Uuid().v4(),
        joinDate = joinDate ?? DateTime.now();
}
